docs/data/data-grid/components/overview.md

```diff
@@ -0,0 +1,94 @@
+# Data Grid - Grid components
+
+<p class="description">Grid components provide a way to extend the default grid UI via composable parts.</p>
+
+## Introduction
+
+Grid components can be used by importing `Grid` from the data grid package:
+
+```tsx
+import { DataGrid, Grid } from '@mui/x-data-grid';
```

Comment:
@mui/xgrid  We might need to think of an alternate name or approach given that there is a Material UI Grid layout component.

A few options I can think of:

1. Rename `Grid` namespace (open to suggestion):

```tsx
import { DataGrid, GridSomething } from "@mui/x-data-grid";

<GridSomething.Toolbar.Root>
  <GridSomething.FilterPanel.Trigger render={<GridSomething.Toolbar.Button />}>
    Filters
  </GridSomething.FilterPanel.Trigger>
</GridSomething.Toolbar.Root>
```

2. Separate import path for new components:

```tsx
import { DataGrid } from "@mui/x-data-grid";
import { GridToolbar, GridFilterPanel } from "@mui/x-data-grid/parts";

<GridToolbar.Root>
  <FilterPanel.Trigger render={<GridToolbar.Button />}>
    Filters
  </FilterPanel.Trigger>
</GridToolbar.Root>
```

3. Different naming convention for new components:

```tsx
import {
  DataGrid,
  DataGridToolbar,
  DataGridFilterPanel,
} from "@mui/x-data-grid";

<DataGridToolbar.Root>
  <DataGridFilterPanel.Trigger render={<DataGridToolbar.Button />}>
    Filters
  </DataGridFilterPanel.Trigger>
</DataGridToolbar.Root>
```

4. New components without namespace (`.` separator):

```tsx
import {
  DataGrid,
  GridToolbarRoot,
  GridToolbarButton,
  GridFilterPanelTrigger,
} from "@mui/x-data-grid";

<GridToolbarRoot>
  <GridFilterPanelTrigger render={<GridToolbarButton />}>
    Filters
  </GridFilterPanelTrigger>
</GridToolbarRoot>
```

Let me know your thoughts, and if you have any other suggestions.

Reply:
I would consider keeping `Grid`, the DX becomes worst with another (longer) name. All our exports already use the `grid` prefix anyway.

Reply:
While `Grid` in option 1 seems confusing to me, I like option 1 the most.